cmd/ipfs/daemon.go

```diff
@@ -488,3 +503,16 @@ func merge(cs ...<-chan error) <-chan error {
 	}()
 	return out
 }
+
+func runGC(req cmds.Request, repo repo.Repo, node *core.IpfsNode) error {
+	disableGC, _, err := req.Option(disableGCKwd).Bool()
+	if err != nil {
+		return err
+	}
+	if disableGC {
+		return nil
+	}
+
+	ctx := req.Context()
+	return fsrepo.PeriodicGC(ctx, repo, node)
```

Comment:
core depends on repo, and `corerepo.GarbageCollect` depends on core, which depends on repo.
Putting the gc function on repo will result in import cycle.

So, back to option 1.